airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java

```diff
@@ -158,17 +158,20 @@ public static Map<AirbyteStreamNameNamespacePair, CursorBasedStatus> getCursorBa
             name);
         final List<JsonNode> jsonNodes = database.bufferedResultSetQuery(conn -> conn.prepareStatement(cursorBasedSyncStatusQuery).executeQuery(),
             resultSet -> JdbcUtils.getDefaultSourceOperations().rowToJson(resultSet));
-        final JsonNode result = jsonNodes.get(0);
         final CursorBasedStatus cursorBasedStatus = new CursorBasedStatus();
-
-        cursorBasedStatus.setStateType(StateType.CURSOR_BASED);
-        cursorBasedStatus.setVersion(2L);
-        cursorBasedStatus.setCursorField(ImmutableList.of(cursorField));
-        cursorBasedStatus.setCursor(result.get(cursorField).asText());
-        cursorBasedStatus.setCursorRecordCount((long) jsonNodes.size());
         cursorBasedStatus.setStreamName(name);
         cursorBasedStatus.setStreamNamespace(namespace);
+        cursorBasedStatus.setCursorField(ImmutableList.of(cursorField));
 
+        if (!jsonNodes.isEmpty()) {
+          final JsonNode result = jsonNodes.get(0);
+          cursorBasedStatus.setStateType(StateType.CURSOR_BASED);
```

Comment:
We can still put version and type on an empty state

Reply:
Done!